File: Graph/2924FindChampionII.py
class Solution:
    def findChampion(self, n: int, edges: List[List[int]]) -> int:
        champion = -1
        weaker = [False] * n

        for edge in edges:
            if not weaker[edge[1]]:
                weaker[edge[1]] = True

        for i in range(n):
            if not weaker[i]:
                if champion == -1:
                    champion = i 
                elif champion != i: 
                    return -1 

        return champion 

File: Graph/2924FindChampionII.go
func findChampion(n int, edges [][]int) int {
	weaker := make([]bool, n)

	for _, v := range edges {
		if !weaker[v[1]] {
			weaker[v[1]] = true
		}
	}

	champion := -1
	for i, v := range weaker {
		if !v {
			if champion != i {
				if champion != -1 {
					return -1
				} else {
					champion = i
				}
			}
		}
	}

	return champion
}
